test/unit/account/states_test.rb

```diff
@@ -213,12 +222,20 @@ def test_schedule_for_deletion
 
       BackendProviderSyncWorker.expects(:enqueue).with(account.id)
 
-      time = Time.zone.now
-      Time.zone.stubs(now: time)
       account.schedule_for_deletion!
-      account.reload
-      assert_equal time.to_s, account.deleted_at.to_s
-      assert_equal time.to_s, account.state_changed_at.to_s
     end
+
+    def test_state_changed_at_from_any_to_any
+      account = FactoryGirl.create(:simple_provider, state: :created)
+
+      %i[make_pending! reject! approve! suspend! resume! schedule_for_deletion!].each do |transition|
+        Timecop.freeze do
+          account.public_send(transition)
+          assert_equal Time.zone.now.to_s, account.reload.state_changed_at.to_s
+          assert_equal Time.zone.now.to_s, account.reload.deleted_at.to_s
```

Comment:
_Idem_ https://github.com/3scale/porta/pull/276/files#r237436761

Reply:
Let's talk about it there, indeed 😄

Reply:
I like it more this way than the stub/mock of `account.expects(:update_attributes).with(...)` because:
1) The test shouldn't care about how it does it, but only about the fact that it does it :slightly_smiling_face: the code could be changed to `update_column` and the test should keep passing
2) The only action that is there before testing the new `state_changed_at` is the action of changing the state, so it is that one that did it :smile:
3) Mocking `update_attributes` doesn't mean testing that it is updated, because there could be errors during the update :slightly_smiling_face:

And there is no need to test that the state is actually updated correctly here because the method being called is the bang one (`!`), so if it doesn't update it, it raises an error.